internal/fsutil/create_test.go

```diff
@@ -93,6 +93,75 @@ var createTests = []createTest{{
 		// mode is not updated.
 		"/foo": "file 0666 d67e2e94",
 	},
+}, {
+	options: fsutil.CreateOptions{
+		Path:         "foo",
+		Mode:         fs.ModeDir | 0775,
+		OverrideMode: true,
+	},
+	hackdir: func(c *C, dir string) {
+		c.Assert(os.Mkdir(filepath.Join(dir, "foo/"), fs.ModeDir|0765), IsNil)
+	},
+	result: map[string]string{
+		// mode is updated.
+		"/foo/": "dir 0775",
+	},
+}, {
+	options: fsutil.CreateOptions{
+		Path:         "foo",
+		Mode:         0775,
+		Data:         bytes.NewBufferString("whatever"),
+		OverrideMode: true,
+	},
+	hackdir: func(c *C, dir string) {
+		err := os.WriteFile(filepath.Join(dir, "foo"), []byte("data"), 0666)
+		c.Assert(err, IsNil)
+	},
+	result: map[string]string{
+		// mode is updated.
+		"/foo": "file 0775 85738f8f",
+	},
+}, {
+	options: fsutil.CreateOptions{
+		Path: "foo",
+		Link: "./bar",
+		Mode: 0666 | fs.ModeSymlink,
+	},
+	hackdir: func(c *C, dir string) {
+		err := os.WriteFile(filepath.Join(dir, "foo"), []byte("data"), 0666)
+		c.Assert(err, IsNil)
+	},
+	result: map[string]string{
+		"/foo": "symlink ./bar",
+	},
```

Comment:
What do you think of adding another test for symlinks with `OverrideMode: true`, to check that for symlinks the mode is not updated?

Reply:
I didn't add this because this is problematic for the following reason:
> chmod never changes the permissions of symbolic links; the chmod system call cannot change their permissions.  This is not a problem  since  the  permissions  of  symbolic  links  are  never used.  However, for each symbolic link listed on the command line, chmod changes the permissions of the pointed file.  In contrast, chmod ignores symbolic links encountered during recursive directory traversals.

What we could do is to check that the mode of the target is not changed, thoughts?